src/FormProvider.jsx

```diff
@@ -1,14 +1,14 @@
 /**
  * Form provider.
- * This provider allows to pass global props to all Form instances
+ * The provider allows to pass global props to all Form instances
  * in order to declare default behaviors on the top level scope.
  */
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 
-const TValidationRules = PropTypes.shape({
-  type: PropTypes.object,
-  name: PropTypes.object
+export const TValidationRules = PropTypes.shape({
+  type: PropTypes.object, // type-specific field validation rules
+  name: PropTypes.object // name-specific field validation rules
 });
 
 export default class FormProvider extends Component {
```

Comment:
Maybe use name `FormContext` because thats what it is :)

Reply:
Usually those things are named ending with `Provider` (`Provider` from Redux, `ApolloProvider` from Apollo). Don't want to break the pattern)